.github/workflows/release.yml

```diff
@@ -0,0 +1,53 @@
+name: Release to NPM and Github
+on: 
+  push:
+    branches: 
+      - master
+      - next
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Initialize Git Config
+        run: |
+          git config user.name "${GITHUB_ACTOR}"
+          git config user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+      - name: Cache node modules
+        id: cache-nodemodules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - name: Install dependencies
+        if: steps.cache-nodemodules.outputs.cache-hit != 'true'
+        run: yarn install
+      - name: Initialize NPM config
+        run: |
+            npm config set registry https://registry.npmjs.com/ 
+            npm config set //registry.npmjs.com/:_authToken=$NPM_TOKEN
+        env: 
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+      - name: Yarn Tests
```

Comment:
Should we add linting here as well?

Reply:
We can skip here since we run on the branch when raising a PR.